### PR TITLE
Load thumbnails for space images instead of the whole file

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Enhancement: Load space images as preview

We've added a new logic which renders space images as preview to minimize data traffic

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23292/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:115

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23309/67/1
The following scenarios passed on retry:
- webUISharingPublicExpire/shareByPublicLinkExpiringLinks.feature:11

File: packages/web-app-files/src/components/SideBar/Actions/SpaceActions.vue [hunk 1]
```diff
@@ -13,7 +13,7 @@
       name="file"
       multiple
       tabindex="-1"
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Actions/SpaceActions.vue:15]: Why multiple? 

I know not part of this PR but while you're at it...

File: packages/web-app-files/src/mixins/spaces/actions/uploadImage.js [hunk 2]
```diff
@@ -38,10 +39,21 @@ export default {
...
       const graphClient = clientService.graphAuthenticated(this.configuration.server, this.getToken)
       const file = ev.currentTarget.files[0]
 
+      if (
+        !ThumbnailService.SupportedMimeTypes.filter((mimeType) =>
+          mimeType.startsWith('image/')
+        ).includes(file.type)
+      ) {
+        return this.showMessage({
+          title: this.$gettext('Failed to upload space image'),
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/uploadImage.js:52]: Please give the user context (`desc` key) why the upload failed. In this case you know that the file type is unsupported. A description like `The file type is unsupported` would be beneficial.

File: packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue [hunk 3]
```diff
@@ -81,12 +83,16 @@ export default {
...
-      spaceImage.value = Buffer.from(fileContents).toString('base64')
+      const fileInfo = yield ref.$client.files.fileInfo(buildWebDavSpacesPath(ref.space.id, path))
+      const resource = buildResource(fileInfo)
+      spaceImage.value = yield loadPreview({
+        resource,
+        isPublic: false,
+        dimensions: ImageDimension.Preview,
+        server: ref.configuration.server,
+        userId: ref.user.id,
+        token: ref.getToken
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue:94]: since you have `isPublic` hardcoded to `false` you can omit the `token` from the options.

Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue]: Discussed this via chat. I had a public link token in mind. Of course the access token is needed for fetching the preview with the authenticated context. Sorry for the noise.

File: packages/web-app-files/src/views/spaces/Project.vue [hunk 4]
```diff
@@ -345,16 +353,20 @@ export default {
...
+
+        this.$client.files.fileInfo(buildWebDavSpacesPath(this.space.id, path)).then((data) => {
+          const resource = buildResource(data)
+          loadPreview({
+            resource,
+            isPublic: false,
+            dimensions: ImageDimension.Preview,
+            server: this.configuration.server,
+            userId: this.user.id,
+            token: this.getToken
```
[the remark this answers is not in the document]
Comment (kulmann) [reply packages/web-app-files/src/views/spaces/Project.vue]: Same here, discussed this via chat. I had a public link token in mind. Of course the access token is needed for fetching the preview with the authenticated context. Sorry for the noise.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23333/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue [hunk 5]
```diff
@@ -174,6 +181,15 @@ export default {
       )
     }
   },
+  watch: {
+    'space.spaceImageData': {
+      handler: function (val) {
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue:186]:
```suggestion
      handler(val) {
```

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23361/60/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:342

File: packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue [hunk 6]
```diff
@@ -174,6 +181,15 @@ export default {
...
+        if (!val) return
```
Comment (dschmidt) [inline packages/web-app-files/src/components/SideBar/Details/SpaceDetails.vue:187]: Not a fan of short syntax for ifs

File: packages/web-app-files/src/mixins/spaces/actions/uploadImage.js [hunk 7]
```diff
@@ -38,10 +39,21 @@ export default {
       this.$data.$_uploadImage_selectedSpace = resources[0]
       this.$refs.spaceImageInput.click()
     },
-    $_uploadImage_uploadImageSpace(ev) {
+    $_uploadImage_uploadImageSpace: function (ev) {
```
Comment (dschmidt) [inline packages/web-app-files/src/mixins/spaces/actions/uploadImage.js:42]: Why this change? I think it's a downgrade hehehehe :trollface:

**Review by dschmidt — COMMENTED** [review]
nice 👍🏻

File: packages/web-app-files/src/mixins/spaces/actions/uploadImage.js (hunk 7 above)
Comment (kulmann) [reply packages/web-app-files/src/mixins/spaces/actions/uploadImage.js:42]: that's weird o_O and I agree with @dschmidt 's initial comment, it's a downgrade 😢

File: packages/web-app-files/src/index.js [hunk 8]
```diff
@@ -138,5 +138,13 @@ export default {
       store.getters.configuration.server || window.location.origin,
       get(store, 'getters.capabilities.files.archivers', [])
     )
+    // FIXME: Remove mock data
+    thumbnailService.initialize(
+      get(store, 'getters.capabilities.files.thumbnail', {
```
Comment (kulmann) [inline packages/web-app-files/src/index.js:143]: If guess we'll name it differently (like `preview` or something more generic), but that's not an issue for now. The way you implemented it allows a fix with minimal footprint ❤️

Comment (kulmann) [reply packages/web-app-files/src/index.js:143]: Let's also check in the future if we can add supported mime types to the capabilities of oc10 as soon as the json schema for the capability is fixed.

**Review by kulmann — APPROVED** [review]
😍

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait2</strong> https://drone.owncloud.com/owncloud/web/23376/45/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/23376/44/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal3</strong> https://drone.owncloud.com/owncloud/web/23378/61/1
The following scenarios passed on retry:
- webUISharingInternalUsersSharingIndicator/shareWithUsers.feature:165

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23378/60/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:342

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/23378/68/1
The following scenarios passed on retry:
- webUISharingPublicManagement/shareByPublicLink.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23378/69/1
The following scenarios passed on retry:
- webUITrashbinDelete/trashbinDelete.feature:143
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUITrashbinRestore-trashbinRestore_feature-L279.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10FileOperations</strong> https://drone.owncloud.com/owncloud/web/23382/14/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.
